### PR TITLE
MTC-250 making db connection string generic and resource monitor sett…

### PR DESCRIPTION
**Pull request by martin8282**
…ing optional

### CODE REVIEW

File: src/Lykke.Job.CandlesHistoryWriter/Startup.cs [hunk 1]
```diff
@@ -89,7 +89,7 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
                     Log = CreateLogWithSlack(
                         services,
                         settings.CurrentValue.SlackNotifications,
-                        candlesHistoryWriter.ConnectionString(x => x.Db.SqlConnectionString),
+                        candlesHistoryWriter.ConnectionString(x => x.Db.SnapshotsConnectionString),
```
Comment (andrey-anokhin) [inline src/Lykke.Job.CandlesHistoryWriter/Startup.cs]: Should be LogsConnectionString

File: src/Lykke.Job.CandlesHistoryWriter/DependencyInjection/JobModule.cs [hunk 2]
```diff
@@ -166,7 +166,7 @@ private void RegisterCandles(ContainerBuilder builder)
                 builder.RegisterType<SqlCandlesHistoryRepository>()
                     .As<ICandlesHistoryRepository>()
                     .WithParameter(TypedParameter.From(_candleHistoryAssetConnections))
-                    .WithParameter(TypedParameter.From(_settings.Db.SqlConnectionString))
+                    .WithParameter(TypedParameter.From(_settings.Db.SnapshotsConnectionString))
```
Comment (andrey-anokhin) [inline src/Lykke.Job.CandlesHistoryWriter/DependencyInjection/JobModule.cs]: Why _settings.Db.SnapshotsConnectionString? It should not have this parameter in constructor at all.

File: src/Lykke.Job.CandlesHistoryWriter.Services/Settings/CandlesHistoryWriterSettings.cs [hunk 3]
```diff
@@ -11,7 +13,10 @@ public class CandlesHistoryWriterSettings
         public DbSettings Db { get; set; }
         public MigrationSettings Migration { get; set; }
```
Comment (andrey-anokhin) [inline src/Lykke.Job.CandlesHistoryWriter.Services/Settings/CandlesHistoryWriterSettings.cs]: Why it is not optional?

File: src/Lykke.Job.CandlesHistoryWriter/Startup.cs (hunk 1 above)
Comment (martin8282) [reply src/Lykke.Job.CandlesHistoryWriter/Startup.cs]: addressed

File: src/Lykke.Job.CandlesHistoryWriter.Services/Settings/CandlesHistoryWriterSettings.cs (hunk 3 above)
Comment (martin8282) [reply src/Lykke.Job.CandlesHistoryWriter.Services/Settings/CandlesHistoryWriterSettings.cs]: it is not valid on declaration type, it is valid only on parameter declaration, and I made the all optional

File: src/Lykke.Job.CandlesHistoryWriter/DependencyInjection/JobModule.cs (hunk 2 above)
Comment (martin8282) [reply src/Lykke.Job.CandlesHistoryWriter/DependencyInjection/JobModule.cs]: addressed as discussed also changed settings

**Pull request merged**